### PR TITLE
Update EnergyStorage.java

### PR DESCRIPTION
**Pull request by mattiamalonni**
Added function for setting energy to custom value (useful for initializations)

### CODE REVIEW

File: src/main/java/net/minecraftforge/energy/EnergyStorage.java [hunk 1]
```diff
@@ -29,6 +29,17 @@ public EnergyStorage(int capacity, int maxReceive, int maxExtract)
...
+    
+    public void setEnergy(int energy)
+    {
+		this.energy = energy;
+
+		if (this.energy > capacity) {
+			this.energy = capacity;
+		} else if (this.energy < 0) {
+			this.energy = 0;
+		}
```
Comment (mezz) [inline src/main/java/net/minecraftforge/energy/EnergyStorage.java]: Be careful not to use tabs, Forge uses spaces and so this is indented strangely.

**Review by mezz — CHANGES_REQUESTED** [review]

**Comment by williewillus** [comment]
You can  use MathHelper.clamp instead of reimplementing its logic

**Comment by liach** [comment]
Please set up a workspace. A pull based on GitHub edits is never merged. Do not tell me you don't know to import classes!

**Comment by MattiaMalonni** [comment]
@liach you should take a look at my last commit...

**Comment by liach** [comment]
Again, MAKE CODE SIMPLE, use net.minecraft.util.math.MathHelper!

**Review by liach — CHANGES_REQUESTED** [review]
You edited this ON GITHUB! That's BAD. Test?

**Comment by MattiaMalonni** [comment]
@liach there is nothing to test, is just a MIN and a MAX (deliberately used to keep consistency with the others methods)

**Comment by LexManos** [comment]
Better to do in the constructor, or use other means of accessing the protected fields, {Hit thats why they are protected not private} We don't want people thinking set is a valid API function.

**Comment by mattiamalonni** [comment]
@LexManos aren't API managed by the interface?

**Comment by LexManos** [comment]
In theory, but you're also expanding the interface with the addition of a new public method which would entice people to use it. Which causes people to make had external deps on internal implementations.  Welcome to the world of developing.

**Comment by BlayTheNinth** [comment]
The change only affected the default implementation (`EnergyStorage`), the interface (`IEnergyStorage`) wasn't actually touched at all.

Having a setEnergyStored method in the default implementation would be useful for syncing as well as loading. So far I've always had to extend EnergyStorage just for that purpose.

**Comment by MattiaMalonni** [comment]
Same here, I actually prefer to create a separate method instead of using the constructor.. But, you know, developers philosophies..

**Comment by LexManos** [comment]
In a perfect world it'd be fine as we would be dealin with real developers who know the difference between internal and external functions. However the 20 years of dealing with these types of communities has proven to me that we can't assume that. Over those years I've developed a sense for when things will get abused and cause much more trouble then it's worth. And this is throwing giant red flags, so we must code defensively.

**Review by mezz — APPROVED** [review]

File: src/main/java/net/minecraftforge/energy/EnergyStorage.java [hunk 2]
```diff
@@ -15,19 +15,25 @@
...
     {
+        this(capacity, maxReceive, maxExtract, 0);
+    }
+
+    public EnergyStorage(int capacity, int maxReceive, int maxExtract, int energy)
+    {
         this.capacity = capacity;
         this.maxReceive = maxReceive;
         this.maxExtract = maxExtract;
+        this.energy = Math.max(0 , Math.min(capacity, energy));
```
Comment (liach) [inline src/main/java/net/minecraftforge/energy/EnergyStorage.java:36]: Why not use MathHelper?

**Review by liach — CHANGES_REQUESTED** [review]

File: src/main/java/net/minecraftforge/energy/EnergyStorage.java (hunk 2 above)
Comment (MattiaMalonni) [inline src/main/java/net/minecraftforge/energy/EnergyStorage.java]: As I previously said, to keep consistency with other methods.. Don't be stubborn..

Comment (liach) [reply src/main/java/net/minecraftforge/energy/EnergyStorage.java:36]: You can say whatever you like here then.

**Review by liach — CHANGES_REQUESTED** [review]
Where is the test mod?

**Pull request merged**